### PR TITLE
[98] [Scaffold] Full Wallet History

### LINKED ISSUE #98: [Scaffold] Full Wallet History
objectives:

The main objective is to show the users the (un-styled) information available for them in their wallet history.

Although the underpinnings of the data layer on the SDK are not defined yet we can Identify certain events or transactions

- Received ZEC (confirmed or pending)
- Sent ZEC (confirmed or pending)
- Transactions that failed to send (failed pending)
- Autoshielding transactions
- Shielded Pool migration transaction (from sapling to orchard)

User stories:

| As   | I would like to                                                                  | So that                                                                        |
|------|----------------------------------------------------------------------------------|--------------------------------------------------------------------------------|
| user | see most recent transactions without leaving my home screen                      | so that I can see latest activity in a glance                                  |
| user | see my whole known history                                                       | so that I can inspect the past transactions involving my keys                  |
| user | see failed transactions                                                          | I can be aware transactions that failed to send                                |
| user | see shield by default transactions                                               | I can keep track of how my wallet has been keeping my funds private            |
| user | see pool migration transactions                                                  | I can keep track of how my wallet has been keeping in the latest shielded pool |
| user | see transactions containing a payment request in the memo as a "payment request" | so that I can quickly attempt to fulfill it if funds available                 |
| user | see transactions that have memos with zero-value as messages                     | I can be aware that there's probably an important memo in that tx              |

Things that remain undefined for now:
- standardized logic to identify autoshielding transactions
- standardized logic to identify pool migration transactions
- "Wallet import" event 
- Whether the history will allow users to see filtered views of their history

Out of Scope:
- definitions of such undefined items above. Please use placeholder functions or variables to mock the appearance until that logic is defined.
- pagination support
- detection of autoshielding or pool migration on real transactions. 

Other Limitations:
the SDK doesn't provide the data needed to show the information to the user on:
- Autoshielding
- pool migration
- previously used outgoing transactions.
- fiat conversion at the time of the transaction
- statistics on accounted ZEC value shielded from transparent address

Those should be mocked and left as placeholders as well.

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #98 

- concept of activities instead of transactions
- the drawer is now fully universal and data driven, rendering some View, going to some detail View
- unit tests work again
- refactor of Activity -> WalletEvent
- WalletEvent view builders simplified
- transactions & wallet events timestamps instead of Date type

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secant/Features/Home/HomeStore.swift [hunk 1]
```diff
@@ -43,12 +43,12 @@ enum HomeAction: Equatable {
...
     case scan(ScanAction)
     case synchronizerStateChanged(WrappedSDKSynchronizerState)
-    case transactionHistory(TransactionHistoryFlowAction)
+    case walletEvents(WalletEventsFlowAction)
     case updateBalance(Balance)
     case updateDrawer(DrawerOverlay)
     case updateRoute(HomeState.Route?)
     case updateSynchronizerStatus
-    case updateTransactions([TransactionState])
+    case updateTransactions([WalletEvent])
```
Comment (pacu) [inline secant/Features/Home/HomeStore.swift:51]: Wouldn't this be...
```suggestion
    case updateWalletEvents([WalletEvent])
```

File: secant/Wrappers/WrappedSDKSynchronizer.swift [hunk 2]
```diff
@@ -160,37 +160,41 @@ class LiveWrappedSDKSynchronizer: WrappedSDKSynchronizer {
...
         
         return .none
     }
     
-    func getAllPendingTransactions() -> Effect<[TransactionState], Never> {
+    func getAllPendingTransactions() -> Effect<[WalletEvent], Never> {
         if let pendingTransactions = try? synchronizer?.allPendingTransactions(),
         let syncedBlockHeight = synchronizer?.latestScannedHeight {
             return Effect(value: pendingTransactions.map {
                 // TODO: - can we initialize it with latestBlockHeight: = nil?
```
Comment (pacu) [inline secant/Wrappers/WrappedSDKSynchronizer.swift:178]: The answer to this question is "yes, you can, but you shouldn't".

we should rely on either latest height or latest scanned height, whatever is available in that order

**Review by pacu — APPROVED** [review]
Looks good to me!

There's one small naming suggestion and a non-blocking comment.

File: secant/Wrappers/WrappedSDKSynchronizer.swift (hunk 2 above)
Comment (LukasKorba) [reply secant/Wrappers/WrappedSDKSynchronizer.swift:178]: true, TODO removed

**Pull request merged**